### PR TITLE
fix(types): allow string for watch handlers

### PR DESCRIPTION
**Pull request by OrkhanAlikhanov**
**What kind of change does this PR introduce?** (check at least one)

- [x] Bugfix
- [ ] Feature
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [x] No

If yes, please describe the impact and migration path for existing applications:

**The PR fulfills these requirements:**

- [x] It's submitted to the `dev` branch for v2.x (or to a previous version branch), _not_ the `master` branch
- [ ] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)
- [x] All tests are passing: https://github.com/vuejs/vue/blob/dev/.github/CONTRIBUTING.md#development-setup
- [x] New/updated tests are included

If adding a **new feature**, the PR's description includes:
- [ ] A convincing reason for adding this feature (to avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it)

**Other information:**

Adds missing TypeScript type info allowing watcher handler to be of `string` type

### CODE REVIEW

File: types/options.d.ts [hunk 1]
```diff
@@ -177,7 +177,7 @@ export interface WatchOptions {
 }
 
 export interface WatchOptionsWithHandler<T> extends WatchOptions {
-  handler: WatchHandler<T>;
+  handler: WatchHandler<T> | string;
```
Comment (posva) [inline types/options.d.ts]: The WatchHandler type itself should include string as it appears in other places

Comment (OrkhanAlikhanov) [reply types/options.d.ts]: Moved it there @posva

**Comment by OrkhanAlikhanov** [comment]
@posva anything on my side to get this merged in?

**Comment by posva** [comment]
No, just waiting :)

**Pull request merged**